src/app/shared-services/db/migrations.ts

```diff
@@ -70,6 +70,10 @@ export const MIGRATIONS = [
     PRIMARY KEY (trajectory, time),
     FOREIGN KEY (trajectory) REFERENCES trajectories(id) ON DELETE CASCADE);`,
 
-  // update date from iso-strings to timestamps
-  `UPDATE points SET time=CAST(strftime('%s', time) AS INT);`,
+  // remove potential duplicate points with very similar timestamp
+  // update date from iso-strings to timestamps and invalidate durationDays for recalculation
+  `DELETE FROM points WHERE time NOT IN (
+    SELECT TIME FROM points GROUP BY strftime('%s', time));
```

Comment:
This inner statement `SELECT TIME FROM points GROUP BY strftime('%s', time)` shows all datetimes in current format, but grouped by the new timestamp-representation - therefore elements, where this timestamp would be the same are ignored and only one is shown. Since other elements with the same timestamp are not in there, those are deleted by the outer statement.

Background:
I came across the case, that I had two very similar datetime-strings in my db, somewhat like `2011-08-12T20:17:46.030Z` and  `2011-08-12T20:17:46.050Z` - slightly different times, but both would share the same timestamp in seconds, something like `1313180266`. Since timestamp is one of our primary-keys, this can't happen. Therefore those multiple objects with the exact same timestamp need to be reduced to one.

Does this make things clearer?

Reply:
Yes, it does, thanks for the clarification. 👍